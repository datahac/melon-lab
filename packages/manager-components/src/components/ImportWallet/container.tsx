import { withFormik } from 'formik';
import { compose } from 'recompose';
import * as Yup from 'yup';
import ImportWallet from './index';

const initialValues = {
  password: '',
};

const withFormValidation = withFormik({
  mapPropsToValues: props =>
    props.formValues ? { ...props.formValues } : initialValues,
  validationSchema: Yup.object().shape({
    password: Yup.string().required('Password is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values),
});

export default compose(withFormValidation)(ImportWallet);
