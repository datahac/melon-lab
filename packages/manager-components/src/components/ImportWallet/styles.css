.import-wallet {
  max-width: 400px;
  margin: 0 auto var(--space-xl) auto;
}

.import-wallet__dropzone {
  margin-bottom: var(--space-m);
}

.import-wallet__input,
.import-wallet__error {
  margin-bottom: var(--space-m);
}
