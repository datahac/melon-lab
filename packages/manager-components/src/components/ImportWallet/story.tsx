import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import ImportWallet from './container';

const data = {
  initialValues: {
    password: '',
  },
  onSubmit: action('onSubmit'),
  onImportFile: action('parseWallet'),
};

storiesOf('Components|Import Wallet', module).add('Default', () => {
  return <ImportWallet {...data} />;
});
