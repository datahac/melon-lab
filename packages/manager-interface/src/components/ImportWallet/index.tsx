import ImportWallet from '~/components/ImportWallet/container';
import { compose, withState, withHandlers } from 'recompose';
import Router from 'next/router';
import WalletMutation from './data/wallet';

const withImportWalletHandlers = withHandlers({
  onImportFile: props => file => {
    const reader = new FileReader();

    reader.onloadend = () => {
      props.setFile(reader.result);
    };

    reader.readAsBinaryString(file[0]);
  },
});

const withImportWalletFileState = withState('file', 'setFile', null);
const withImportWalleteErrorState = withState('error', 'setError', null);

const redirect = () =>
  Router.replace({
    pathname: '/wallet',
  });

const withImportWallet = BaseComponent => baseProps => (
  <WalletMutation onCompleted={redirect}>
    {(importWallet, walletProps) => (
      <BaseComponent
        onImportFile={baseProps.onImportFile}
        file={baseProps.file}
        serverError={baseProps.error}
        onSubmit={values =>
          importWallet({
            variables: {
              file: baseProps.file,
              password: values.password,
            },
          }).catch(error => baseProps.setError(error.message))
        }
        loading={walletProps.loading}
      />
    )}
  </WalletMutation>
);

export default compose(
  withImportWalletFileState,
  withImportWalleteErrorState,
  withImportWalletHandlers,
  withImportWallet,
)(ImportWallet);
