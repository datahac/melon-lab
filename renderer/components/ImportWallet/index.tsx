import React, { useState } from 'react';
import ImportWallet from '~/components/ImportWallet';
import Composer from 'react-composer';
import { withRouter } from 'next/router';
import WalletMutation from './data/wallet';
import withForm from './withForm';

const ImportWalletForm = withForm(ImportWallet);

const ImportWalletContainer = ({ router }) => {
  const [file, setFile] = useState(null);
  const [error, setError] = useState(null);

  const onImportFile = file => {
    const reader = new FileReader();
    reader.onloadend = () => {
      if (reader.result !== 'null') {
        setFile(reader.result);
      }
    };

    reader.readAsBinaryString(file[0]);
  };

  return (
    <Composer
      components={[
        ({ render }) => (
          <WalletMutation
            onCompleted={() => {
              router.push({
                pathname: '/wallet',
              });
            }}
          >
            {(a, b) => render([a, b])}
          </WalletMutation>
        ),
      ]}
    >
      {([[importWallet, walletProps]]) => (
        <ImportWalletForm
          onImportFile={onImportFile}
          file={file}
          serverError={error}
          onSubmit={values =>
            importWallet({
              variables: {
                file,
                password: values.password,
              },
            }).catch(setError)
          }
          loading={walletProps.loading}
        />
      )}
    </Composer>
  );
};

export default withRouter(ImportWalletContainer);
