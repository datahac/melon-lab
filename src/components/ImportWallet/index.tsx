import React from 'react';
import ImportWallet from '~/components/ImportWallet';
import Composer from 'react-composer';
import { withRouter } from 'next/router';
import WalletMutation from './data/wallet';
import withForm from './withForm';

const ImportWalletForm = withForm(ImportWallet);

class ImportWalletContainer extends React.Component {
  state = {
    file: null,
    error: null,
  };

  setFile = file => {
    this.setState({ file });
  };

  setError = error => {
    this.setError({ error });
  };

  onImportFile = file => {
    const reader = new FileReader();
    reader.onloadend = () => {
      this.setFile(reader.result);
    };

    reader.readAsBinaryString(file[0]);
  };

  render() {
    return (
      <Composer
        components={[
          ({ render }) => (
            <WalletMutation
              onCompleted={() => {
                this.props.router.replace({
                  pathname: '/wallet',
                });
              }}
            >
              {(a, b) => render([a, b])}
            </WalletMutation>
          ),
        ]}
      >
        {([[importWallet, walletProps]]) => (
          <ImportWalletForm
            onImportFile={this.onImportFile}
            file={this.state.file}
            serverError={this.state.error}
            onSubmit={values =>
              importWallet({
                variables: {
                  file: this.state.file,
                  password: values.password,
                },
              }).catch(this.setError)
            }
            loading={walletProps.loading}
          />
        )}
      </Composer>
    );
  }
}

export default withRouter(ImportWalletContainer);
