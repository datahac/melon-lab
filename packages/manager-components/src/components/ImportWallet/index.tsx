import React, { StatelessComponent } from 'react';
import Dropzone from 'react-dropzone';
import Button from '~/blocks/Button';

import styles from './styles.css';

export interface ImportWalletProps {
  goToAccount: () => void;
  parseWallet: () => void;
}

export const ImportWallet: StatelessComponent<ImportWalletProps> = ({
  goToAccount,
  parseWallet,
}) => {
  return (
    <div className="import-wallet">
      <style jsx>{styles}</style>
      <h3>Import wallet</h3>
      <div className="import-wallet__dropzone">
        <Dropzone
          onDrop={parseWallet}
          style={{
            width: '100%',
            border: '1px dotted black',
            padding: 10,
            textAlign: 'center',
            cursor: 'pointer',
          }}
        >
          <p>Select a wallet JSON file from your computer</p>
        </Dropzone>
      </div>
      <Button style="secondary" onClick={goToAccount}>
        Cancel
      </Button>
    </div>
  );
};

export default ImportWallet;
