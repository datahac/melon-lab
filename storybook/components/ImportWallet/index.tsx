import React, { Fragment, StatelessComponent } from 'react';
import Dropzone from 'react-dropzone';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import Spinner from '~/blocks/Spinner';
import Notification from '../../blocks/Notification';
import Link from '~/blocks/Link';

import styles from './styles.css';

interface FormValues {
  password: string;
}

export interface ImportWalletProps {
  errors?: any;
  handleBlur?: () => void;
  handleSubmit?: () => void;
  handleChange?: () => void;
  touched?: any;
  values: FormValues;
  loading?: boolean;
  onImportFile: () => void;
  file?: string;
  serverError?: string;
}

export const ImportWallet: StatelessComponent<ImportWalletProps> = ({
  errors,
  handleBlur,
  handleSubmit,
  handleChange,
  touched,
  values,
  loading,
  onImportFile,
  file,
  serverError,
}) => (
  <div className="import-wallet">
    <style jsx>{styles}</style>
    {serverError && (
      <div className="import-wallet__error">
        <Notification isError>{serverError}</Notification>
      </div>
    )}
    {loading ? (
      <Spinner icon size="small" text="Import Wallet..." />
    ) : (
      <Fragment>
        {!file ? (
          <div className="import-wallet__dropzone">
            <Dropzone
              onDrop={onImportFile}
              style={{
                width: '100%',
                border: '1px dotted black',
                padding: 10,
                textAlign: 'center',
                cursor: 'pointer',
              }}
            >
              <p>Select a wallet JSON file from your computer</p>
            </Dropzone>
          </div>
        ) : (
          <Form onSubmit={handleSubmit}>
            <div className="import-wallet__input">
              <Input
                maxlength={64}
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.password}
                required={true}
                name="password"
                type="password"
                placeholder="Password"
                error={touched.password && errors.password}
              />
            </div>
            <div className="import-wallet__actions">
              <div className="import-wallet__action">
                <Link
                  style="secondary"
                  size="medium"
                  href={{
                    pathname: '/wallet',
                  }}
                >
                  Cancel
                </Link>
              </div>
              <div className="import-wallet__action">
                <Button type="submit" style="primary">
                  Import
                </Button>
              </div>
            </div>
          </Form>
        )}
      </Fragment>
    )}
  </div>
);

export default ImportWallet;
