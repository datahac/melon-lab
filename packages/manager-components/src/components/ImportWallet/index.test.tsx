import React from 'react';
import ImportWallet from './index';

const data = {};

describe('ImportWallet', () => {
  const defaultElement = <ImportWallet {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
