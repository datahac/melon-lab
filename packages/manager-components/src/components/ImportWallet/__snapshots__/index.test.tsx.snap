// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportWallet should render correctly 1`] = `
<div
  className="jsx-__scopedHash import-wallet"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Import wallet
  </h3>
  <div
    className="jsx-__scopedHash import-wallet__error"
  />
  <div
    className="jsx-__scopedHash import-wallet__dropzone"
  >
    <t
      disableClick={false}
      disablePreview={false}
      disabled={false}
      getDataTransferItems={[Function]}
      inputProps={Object {}}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      preventDropOnDocument={true}
      style={
        Object {
          "border": "1px dotted black",
          "cursor": "pointer",
          "padding": 10,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      <p
        className="jsx-__scopedHash"
      >
        Select a wallet JSON file from your computer
      </p>
    </t>
  </div>
</div>
`;
